packages/realm-web/src/transports/AuthenticatedTransport.ts

```diff
@@ -0,0 +1,48 @@
+import { Transport, Request } from "./Transport";
+import { PrefixedTransport } from "./PrefixedTransport";
+
+interface UserContext {
```

Comment:
why do we need a type for this

Reply:
Because I wanted the authenticated transport to have a looser coupling to the App.

Reply:
I did not clarify enough. `UserContext` is having only a `curerntUser`. Why not pass currentUser where needed and remove the `UserContext` type.

Reply:
ok I downloaded the branch and checked the whole tranports code. it's clear to me now. thanks.

Reply:
Seems to me that a `PrefixedTransport` can be created from an `AuthenticatedTransport` but then instance does not use the userContext when doing  a fetch. Is this intentional ?

Reply:
The `PrefixedTransport` wraps another `Transport` instance, through the first argument on its constructor (in this case, the `AuthenticatedTransport` instance which holds the reference to the `userContext`). Why would the `AuthenticatedTransport` not use the `userContext` then?

Reply:
Maybe I am wrong but I mean this https://github.com/realm/realm-js/pull/2741/files#r412911309